src/main/java/org/fenixedu/academic/thesis/ui/service/ThesisProposalsService.java

```diff
@@ -199,6 +200,13 @@ public ThesisProposal createThesisProposal(ThesisProposalBean proposalBean, Stri
             throw new UnexistentThesisParticipantException();
         }
 
+        User thesisCreator = Authenticate.getUser();
+        if (!proposalBean.getExecutionDegreeSet().stream()
+                .anyMatch(e -> CoordinatorGroup.get(e.getDegree()).isMember(thesisCreator))
+                && !participants.stream().map(bean -> bean.getUser()).collect(Collectors.toSet()).contains(thesisCreator)) {
```

Comment:
creating a new set just to do a contains invocation is not ok with the efficiency gods